components/databaseHandler.ts

```diff
@@ -216,33 +217,48 @@ export async function writeExternalUserData(
 /**
  * Reads a contract session from the contractSessions folder.
  *
- * @param sessionId The ID of the session to load.
+ * @param identifier The identifier for the saved session, in the format of token_sessionID.
  * @returns The contract session.
  */
 export async function getContractSession(
-    sessionId: string,
+    identifier: string,
 ): Promise<ContractSession> {
+    const files = await readdir("contractSessions")
+    const filtered = files.filter((fn) => fn.endsWith(`_${identifier}.json`))
+    if (filtered.length === 0) {
+        throw Error(`No session saved with identifier ${identifier}`)
+    }
+    // The filtered files have the same identifier, they are just stored at different slots
+    // So we can read any of them and it will be the same.
```

Comment:
```suggestion
    const filtered = files.filter((fn) => fn.endsWith(`_${identifier}.json`))

    if (filtered.length === 0) {
        throw new Error(`No session saved with identifier ${identifier}`)
    }

    // The filtered files have the same identifier, they are just stored at different slots
    // So we can read any of them and it will be the same.
```